hal/armv7a/zynq7000/timer.c

```diff
@@ -92,38 +76,9 @@ static void timer_setPrescaler(u32 freq)
 }
 
 
-int timer_wait(u32 ms, int flags, volatile u16 *p, u16 v)
+time_t hal_getTime(void)
 {
-	/* Set value that determines when an irq t will be generated */
-	timer_common.leftTime = ms;
-	*(timer_common.base + interval_val) |= timer_common.ticksPerFreq & 0xffff;
-
-	/* Reset counter */
-	*(timer_common.base + cnt_ctrl) = 0x2;
-	/* Enable interval irq timer */
-	*(timer_common.base + ier) = 0x1;
-
-	while (timer_common.leftTime) {
-		if ((flags & TIMER_VALCHG) && *p != v)
-			return 1;
-	}
-
-	return 0;
-}
-
-
-void timer_init(void)
-{
-	timer_common.irq = TTC0_1_IRQ;
-	timer_common.base = TTC0_BASE_ADDR;
-
-	/* Reset controller */
-	timer_done();
-
-	/* Trigger interrupt at TTC_DEFAULT_FREQ = 1000 Hz */
-	timer_setPrescaler(TTC_DEFAULT_FREQ);
-
-	hal_irqinst(timer_common.irq, timer_isr, (void *)&timer_common);
+	return timer_common.time;
```

Comment:
This needs to be done in the critical section (with irq disabled).

Reply:
done